ddprofiler/app/src/test/java/ddprofiler/test/PreAnalyzerTest.java

```diff
@@ -32,6 +46,83 @@ public class PreAnalyzerTest {
     private String username = "root";
     private String password = "Qatar";
 
+    /**
+     * Setup PreAnalyzer before each test
+     * (specifically for new tests, not legacy ones).
+     */
+    @Before
+    public void setUp() throws CsvValidationException, SQLException, IOException {
+        // Avoid Exceptions by adding a sample column to the source.
+        when(source.getAttributes()).thenReturn(new ArrayList<>(Arrays.asList(new Attribute("sample"))));
```

Comment:
(Also valid for similar instantiations) new ArrayList<>(Arrays.asList()) can be replaced with List.of() which is more concise.

Reply:
I see, I will adopt the changes as it is way cleaner and easier to read as well.